File: src/db-gwent-server-scala/src/main/scala/com/entities/abilities/RowCardsToOne_Ability.scala
package com.entities.abilities

import com.entities.{CardRowType, Player}

case class RowCardsToOne_Ability(
  row: CardRowType)
  extends Ability:

  def apply(player: Player, enemy: Player): Boolean =
    Array(player, enemy).forall: p =>
      p.battleCardsTo1(row, RowCardsToOne_Ability.Id)

object RowCardsToOne_Ability:
  val Id = "row-cards-to-1"

File: src/db-gwent-server-kotlin/src/main/kotlin/com/entities/abilities/RowCardsToOne_Ability.kt
package com.entities.abilities

import com.entities.Card
import com.entities.Player
import com.entities.RowType

class RowCardsToOne_Ability(
    val row: RowType) : Ability {

    override fun apply(card: Card, player: Player, enemy: Player): Boolean =
        arrayOf(player, enemy).all{
            it.battleCardsTo1(row, RowCardsToOne_Ability.Id)
        }

    companion object {
        val Id = "row-cards-to-1"
    }

}
